src/ws-dropdown.js

```diff
@@ -0,0 +1,252 @@
+    items: React.PropTypes.array,
+    noTrigger: React.PropTypes.bool,
+    multiple: React.PropTypes.bool,
+    filterable: React.PropTypes.bool,
+    filter: React.PropTypes.string,
+    limit: React.PropTypes.number,
+    orientation: React.PropTypes.oneOf(['left', 'right'])
+  };
+
+  /**
+   * @type {Object}
+   */
+  static childContextTypes = {
+    multiple: React.PropTypes.bool
+  };
+
+  /**
+   * @param props {Object}
+   * @constructor
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      opened: false,
+      height: 0,
+      text: props.text || props.value,
+      value: this.enrichItems(Array.isArray(props.value) ? props.value : props.value ? [props.value] : []),
+      items: this.enrichItems(props.items)
+    };
+    // Set states to items in item list for passed values
+    this.state.items.forEach(item => {
+      if (this.state.value.find(val => val.label == item.label)) {
+        item.selected = true;
+        item.stored = true;
+      }
+    });
+  }
+
+  /**
+   * Called by React to get the types of the child context values
+   * @returns {Object}
+   */
+  getChildContext() {
+    return {
+      multiple: this.props.multiple
+    }
+  }
+
+  /**
+   * Start listening for clicks in window
+   * @void
+   */
+  componentDidMount() {
+    window.addEventListener('click', this.onDocumentClick.bind(this));
+  }
+
+  /**
+   * Stop listening for clicks in window
+   * @void
+   */
+  componentWillUnmount() {
+    window.removeEventListener('click', this.onDocumentClick.bind(this));
+  }
+
+  /**
+   * Used to convert the items if they are strings into the required object structure
+   * @param items {Array<String|Object>}
+   * @returns {Array<Object>}
+   */
+  enrichItems(items) {
+    return items.map(item => {
+      let enriched = typeof item === 'object' ? item : {label: item};
+      if (enriched.children) {
+        enriched.children = this.enrichItems(enriched.children);
+      }
+      return enriched;
+    });
+  }
+
+  /**
+   * Handles data propagation from child elements
+   * @param type {String} Either change for value changes or change-size which will be emitted on menu changes
+   * @param data {Object|Number} Either new value or height of new menu
+   * @void
+   */
+  handlePropagation(type, data) {
+    if (type === 'change') {
+      this.close();
+      this.setValue(data);
+      this.element.dispatchEvent(new CustomEvent('change', { detail: data }));
+    }
+    else if (type === 'change-size') {
+      this.adjustSize(data);
+    }
+  }
+
+  /**
+   * Set the value of the dropdown and update the display text if the trigger element is a select
+   * @param value {Object|Array<Object>}
+   * @void
+   */
+  setValue(value) {
+    this.state.value = value;
+    // Check if we have to update the text value
+    if (this.props.type === 'select') {
+      if (Array.isArray(this.state.value)) {
+        this.state.text = this.state.value.map(item => item.label).join(', ');
```

Comment:
I'd assign the text value to variable and then setState ex. `this.setState({text});`